### PR TITLE
Added new procedure "Permission Set Relation".AddNewPermissionSetRelation

### PR DESCRIPTION
**Pull request by passion4dev**
Fixes https://github.com/microsoft/ALAppExtensions/issues/21197

### CODE REVIEW

**Comment by passion4dev** [comment]
@microsoft-github-policy-service agree [company="WSB Solutions B.V."]

**Comment by passion4dev** [comment]
@microsoft-github-policy-service agree company="WSB Solutions BV"

File: Modules/System/Permission Sets/src/PermissionSetRelation.Codeunit.al [hunk 1]
```diff
@@ -10,6 +10,24 @@ codeunit 9855 "Permission Set Relation"
...
+    /// <param name="RelatedAppId">The app ID of the related permission set.</param>
+    /// <param name="RelatedRoleId">The role ID of the related permission set.</param>
+    /// <param name="RelatedScope">The scope of the related permission set.</param>
+    /// <param name="PermissionType">The type of the new permission set relation.</param>
+    /// <returns>true if creating the permission set relation was successful</returns>
+    procedure AddNewPermissionSetRelation(AppId: Guid; RoleId: Code[30]; Scope: Option System,Tenant; RelatedAppId: Guid; RelatedRoleId: Code[30]; RelatedScope: Option System,Tenant; PermissionType: Option Include,Exclude) Success: Boolean
+    var
+        PermissionSetRelationImpl: Codeunit "Permission Set Relation Impl.";
+    begin
+        exit(PermissionSetRelationImpl.AddNewPermissionSet(AppId, RoleId, Scope, RelatedAppId, RelatedRoleId, RelatedScope, PermissionType));
```
Comment (darjoo) [inline Modules/System/Permission Sets/src/PermissionSetRelation.Codeunit.al:28]: Calling this function directly bypasses the security checks of VerifyUserCanEditPermissionSet.

Doesn't calling ModifyPermissionSet achieve the same result you're looking for?

Comment (passion4dev) [reply Modules/System/Permission Sets/src/PermissionSetRelation.Codeunit.al]: Hi Darrick,
Thanks your review with a sharp eye.
Unfortunately ModifyPermissionSet does not achieve the same result I'm looking for.

I created a new procedure AddNewPermissionSetRelation in the "Permission Set Relation Impl.".
According the existing pattern the "Permission Set Relation" AddNewPermissionSetRelation now calls "Permission Set Relation Impl." AddNewPermissionSetRelation.

**Review by darjoo — COMMENTED** [review]
Just the minor change to put local back and all is good imo.

File: Modules/System/Permission Sets/src/PermissionSetRelationImpl.Codeunit.al [hunk 2]
```diff
@@ -248,7 +255,7 @@ codeunit 9856 "Permission Set Relation Impl."
             until MetaDataPermissionSetRel.Next() = 0;
     end;
 
-    local procedure AddNewPermissionSet(CurrAppId: Guid; CurrRoleID: Code[30]; CurrScope: Option System,Tenant; RelatedAppId: Guid; RelatedRoleId: Code[30]; RelatedScope: Option System,Tenant; PermissionType: Option Include,Exclude) Success: Boolean
+    procedure AddNewPermissionSet(CurrAppId: Guid; CurrRoleID: Code[30]; CurrScope: Option System,Tenant; RelatedAppId: Guid; RelatedRoleId: Code[30]; RelatedScope: Option System,Tenant; PermissionType: Option Include,Exclude) Success: Boolean
```
[the remark this answers is not in the document]
Comment (passion4dev) [reply Modules/System/Permission Sets/src/PermissionSetRelationImpl.Codeunit.al]: Indeed!
Just fixed it.
Thanks.

**Review by darjoo — APPROVED** [review]

**Review by JesperSchulz — APPROVED** [review]

**Comment by JesperSchulz** [comment]
Let's get this processed.

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 52176.

**Comment by JesperSchulz** [comment]
Availability update: We will publish a fix for this issue in the next update for release 21.

Build ID to track: 52214.

**Pull request merged**